crates/components-runtime/src/data_collection/payload.rs

```
@@ -106,6 +108,292 @@ impl Event {
+                                    if data.name == manipulation.from_property {
+                                        data.name = manipulation.to_property.clone();
+                                    }
+                                } else {
+                                    let value = data.properties.get(&manipulation.from_property);
+                                    if let Some(value) = value {
+                                        data.properties.insert(
+                                            manipulation.to_property.clone(),
+                                            value.clone(),
+                                        );
+                                        data.properties.remove(&manipulation.from_property);
+                                    }
+                                }
+                            });
+                        }
+                    }
+                    "user" => {
+                        if let Data::User(ref mut data) = self.data {
+                            rule.manipulations.iter().for_each(|manipulation| {
+                                let value = data.properties.get(&manipulation.from_property);
+                                if let Some(value) = value {
+                                    data.properties
+                                        .insert(manipulation.to_property.clone(), value.clone());
+                                    data.properties.remove(&manipulation.from_property);
+                                }
+                            });
+                        }
+                    }
+                    _ => {}
+                });
+        });
+    }
+}
+
+pub fn evaluate_boolean_filter(field_value: bool, operator: &str, condition_value: bool) -> bool {
+    match operator {
+        "eq" => field_value == condition_value,
+        "neq" => field_value != condition_value,
+        _ => false,
```

if a wrong operator is used, we might want to log it somewhere because it's probably a bug in the UI or somewhere else :) never fail silently, or this will be un-debuggable in the future